activesupport/lib/active_support/values/time_zone.rb

```diff
@@ -248,7 +249,18 @@ def [](arg)
       # A convenience method for returning a collection of TimeZone objects
       # for time zones in the USA.
       def us_zones
-        @us_zones ||= all.find_all { |z| z.name =~ /US|Arizona|Indiana|Hawaii|Alaska/ }
+        country_zones(:us)
+      end
+
+      # A convenience method for returning a collection of TimeZone objects
+      # for time zones in the country specified by its ISO 3166-1 Alpha2 code.
+      def country_zones(country_code)
+        code = country_code.to_s.upcase
+        @country_zones[code] ||=
+          TZInfo::Country.get(code).zone_identifiers.map do |tz_id|
```

Comment:
will this return a `ThreadSafe::Cache` ? If not, we need to use a each here instead of a map, and push things to our `ThreadSafe::Cache` instead.